### PR TITLE
remove allExportedDestRules

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
The `proxy` passed to push.DestinationRule can not be nil as `service`.

### CODE REVIEW

File: pilot/pkg/model/push_context.go [hunk 1]
```diff
@@ -507,17 +507,13 @@ func NewPushContext() *PushContext {
 		privateVirtualServicesByNamespace: map[string][]Config{},
 		namespaceLocalDestRules:           map[string]*processedDestRules{},
 		namespaceExportedDestRules:        map[string]*processedDestRules{},
-		allExportedDestRules: &processedDestRules{
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/push_context.go]: only removed this

File: pilot/pkg/model/push_context.go [hunk 2]
```diff
@@ -93,7 +93,6 @@ type PushContext struct {
 	// the dest rule based on the most specific host match, and not just any destination rule
 	namespaceLocalDestRules    map[string]*processedDestRules
 	namespaceExportedDestRules map[string]*processedDestRules
-	allExportedDestRules       *processedDestRules
```
Comment (ramaraochavali) [inline pilot/pkg/model/push_context.go]: Can you also remove the comment above this line?

Comment (hzxuzhonghu) [reply pilot/pkg/model/push_context.go]: done

**Comment by hzxuzhonghu** [comment]
There was a conflict with master, rebased

**Pull request merged**